File: CTPS-LAB/18-9-24/eb.cc
// 18-9-24

#include<bits/stdc++.h>
using namespace std;

int main() {
    int myUnit;
    cout << "Enter your unit: ";
    cin >> myUnit;

    int unitSpace = 100;

    int rs1 = 1;
    int rs2 = 1.5;
    int rs3 = 3;
    int rs4 = 5;

    int cost1 = unitSpace * rs1;
    int cost2 = unitSpace * rs2;
    int cost3 = unitSpace * rs3;
    int cost4 = unitSpace * rs4;

    int totalCost = 0;

    if(myUnit <= 0) {
        cout << "Invalid unit. Please enter a value greater than 0." << endl;
    } else if(myUnit <= 100){
        totalCost = cost1;
    }
    else if (myUnit <= 200){
        totalCost = cost1 + cost2;
    }
    else if (myUnit <= 300){
        totalCost = cost1 + cost2 + cost3;
    }
    else {
        totalCost = cost1 + cost2 + cost3 + cost4;
        int extraUnits = myUnit - 300;
        totalCost += (extraUnits / 100) * cost4;
        if(extraUnits % 100 != 0) {
            totalCost += (extraUnits % 100) * rs4;
        }
    }

    cout << totalCost << endl;

    return 0;
}

File: CTPS-LAB/23-9-24/eb.py
# 18-9-24

def calculate_total_cost():
    my_unit = float(input("Enter your unit: "))

    unit_space = 100

    rs1 = 1
    rs2 = 1.5
    rs3 = 3
    rs4 = 5

    cost1 = unit_space * rs1
    cost2 = unit_space * rs2
    cost3 = unit_space * rs3
    cost4 = unit_space * rs4

    total_cost = 0

    if my_unit <= 0:
        print("Invalid unit. Please enter a value greater than 0.")
    elif my_unit <= 100:
        total_cost = cost1
    elif my_unit <= 200:
        total_cost = cost1 + cost2
    elif my_unit <= 300:
        total_cost = cost1 + cost2 + cost3
    else:
        total_cost = cost1 + cost2 + cost3 + cost4
        extra_units = my_unit - 300
        total_cost += (extra_units // 100) * cost4
        if extra_units % 100 != 0:
            total_cost += (extra_units % 100) * rs4

    print("Total cost: ", total_cost)

calculate_total_cost() 

